invenio_rdm_records/services/schemas/access.py

```diff
@@ -33,26 +31,22 @@ class Agent(Schema):
     user = fields.Integer(required=True)
 
 
+class EmbargoSchema(Schema):
+    """Schema for an embargo on the record."""
+
+    # TODO: check that ('until' is in the future <=> 'active' == true)
+    active = fields.Bool()
+    until = ISODateString()
+    reason = SanitizedUnicode()
+
+
 class AccessSchema(Schema):
     """Access schema."""
 
-    metadata = fields.Bool(required=True)
-    files = fields.Bool(required=True)
+    # TODO: check if 'record' and 'files' each are
+    #       one of ['public', 'restricted']
+    record = SanitizedUnicode(required=True)
+    files = SanitizedUnicode(required=True)
     owned_by = fields.List(fields.Nested(Agent))
-    access_right = SanitizedUnicode(required=True)
-    embargo_date = ISODateString()
-    access_condition = fields.Nested(AccessConditionSchema)
-
-    @validates('embargo_date')
-    def validate_embargo_date(self, value):
-        """Validate that embargo date is in the future."""
-        if arrow.get(value).date() <= arrow.utcnow().date():
-            raise ValidationError(
-                _('Embargo date must be in the future.'),
-                field_names=['embargo_date']
-            )
-
-    @validates_schema
-    def validate_access_right(self, data, **kwargs):
-        """Validate that access right is one of the allowed ones."""
-        validate_entry('access_right', data)
+    embargo = fields.Nested(EmbargoSchema)
+    grants = fields.List(fields.Nested(Grant))
```

Comment:
Minor: This one should be disabled for the current release. I don't think we'll have the time to properly check it (at least we should do it as an add-on once the other stuff is working).